### PR TITLE
draft explanation of CORS in spartacus

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Contributes to #545

### CODE REVIEW

File: _pages/dev/cors.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Cross-Origin Resource Sharing (CORS)
+
+While CORS is a standard mechanism on the web, it's considered rather complicated and blocks new Spartacus developer often during their first setup. This is why we spend an article on explaing _how_ CORS affects Spartacus, and _what_ is required to enable the right CORS setup.
```
Comment (hackergil) [inline _pages/dev/cors.md:3]: I've seen that most people don't really know what CORS is. Perhaps add a bit more of explanation about it or some links?

File: _pages/dev/cors.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+The response headers are dictated by the server, which is why the server must be setup to generate the right headers. In the SAP commerce backend, those headers are configurable in a generic fashion, using a CorsFilter. Project properties can be used to configure this for each node, or an installation script (ImpEx) can be used to install to each node. See for more information on the setup in the section below.
+
+## CORS Headers
+
+There are various CORS headers that come into play to specify whether the origin is allowed with or without certain methods, headers, cookies, etc. Each of the configurations and it's use in spartacus is discussed in the following sections.
+
+### allowedOrigins
+
+In development the allowed origins are often configured with a `*`, meaning every client is allowed. In production this should be more precise, to avoid abuse of the API.
```
Comment (hackergil) [inline _pages/dev/cors.md:17]: Perhaps indicate that this acts as a "whitelist" for requests from other domains?

Comment (hackergil) [reply _pages/dev/cors.md:17]: It'd also be good to have a link to CX docs around these

Comment (tobi-or-not-tobi) [reply _pages/dev/cors.md]: done

File: _pages/dev/cors.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+Only Context-Driven Services requires a custom header to be exposed, which is the `x-anonymous-consents` header.
+
+### allowCredentials
+
+Request credentials are concerned with cookies, authorization headers, or TLS client certificates. These are not allowed in cross-origin requests by default, which is why a request is blocked when the configuration is not applied.
+
+Spartacus did not send cookies in version 1.x of the Spartacus libraries, but since version 2.0, cookies will be sent for each and every OCC request. This has also been patched to versions 1.4 and 1.5 of the Spartacus libraries.
+
+Sending cookies is required to gain "session infinity", which is also know as "sticky sessions". Session infinity means that the same server behind an API endpoint is used for all subsequent requests for the same session. Although the Commerce API is stateless, there are occasions when multiple parallel calls could fail (for example, add to cart vs get cart). Moreover, the back end will gain performance improvements if requests for the same session are served by the same server.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/cors.md:54]: We can change the last bit to:

> Although the Commerce API is stateless, there are occasions when multiple parallel or sequential calls could fail. For example, an “add to cart” request followed by a “load cart” request could fail, as the first request could end up at server 1, where as the 2nd request immediately followed after could end up at server 2. The servers might not be fast enough to send around cache invalidation, which is why the 2nd response might fail to catch the added item.

Comment (gladius-mtl) [reply _pages/dev/cors.md]: This is perfect! Thank you very much!! 🙂

**Pull request merged**